lib/modules/ai_bot/site_settings_extension.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module DiscourseAi::AiBot::SiteSettingsExtension
+  module ClassMethods
+    def ai_bot_enabled=(val)
+      super(val)
+      DiscourseAi::AiBot::SiteSettingsExtension.enable_or_disable_ai_bots
+    end
+
+    def ai_bot_enabled_chat_bots=(val)
+      super(val)
+      DiscourseAi::AiBot::SiteSettingsExtension.enable_or_disable_ai_bots
+    end
+  end
+
+  def self.enable_or_disable_ai_bots
+    enabled_bots = SiteSetting.ai_bot_enabled_chat_bots.split("|")
```

Comment:
You can do `SiteSetting.ai_bot_enabled_chat_bots_map` here to get the array :)

Reply:
Could also just do `enabled_bots = SiteSetting.ai_bot_enabled ? SiteSetting.ai_bot_enabled_chat_bots_map : []`